### DISCUSSION TITLE
How can i ensure that my assets and ops retry if something goes wrong during materialization?

### DISCUSSION THREAD

**Discussion by gibsondan** [dagster-io/dagster#16289] [Q&A] [answered]
Dagster allows you to set a RetryPolicy on your assets and ops that ensure that it retries if an Exception is raised during materialization. See https://docs.dagster.io/concepts/ops-jobs-graphs/op-retries#op-retries for more information.

However, a RetryPolicy on an asset or op will not execute if one of the Dagster processes running your code crashes. Dagster runs each happen in an isolated process (the "run worker") and in most cases, each op or asset executes in its own process as well (the "step worker"). If either a run worker or a step worker crashes (for example, due to running out of memory), the RetryPolicy will not kick in.

To ensure that your ops and assets execute even after a crash, you can set up [run-level retries](https://docs.dagster.io/deployment/run-retries#run-retries), which will retry the run if it fails for any reason.

To ensure that your run-level retries kick in after a crash, you may want to enable [Run Monitoring](https://docs.dagster.io/deployment/run-monitoring#run-monitoring) to ensure that your run is moved into a failure state even if the run worker crashes. Run Monitoring is enabled by default in Dagster Cloud.

**Comment by gibsondan** [comment, marked as answer]
Post description has the answer!

**Comment by sryza** [comment]
@gibsondan I'm reopening this because "Closed" questions weirdly don't show up in search results.
